internal/workflow/uuid.go

```diff
@@ -0,0 +1,14 @@
+package workflow
+
+import (
+	"github.com/google/uuid"
+	temporalsdk_workflow "go.temporal.io/sdk/workflow"
+)
+
+func GenUUID(ctx temporalsdk_workflow.Context) (uuid.UUID, error) {
+	var id uuid.UUID
+	gen := func(ctx temporalsdk_workflow.Context) interface{} { return uuid.New() }
+	enc := temporalsdk_workflow.SideEffect(ctx, gen)
```

Comment:
Sadly, https://github.com/temporalio/sdk-go/issues/916. That's why I went with an UUID generator.

Reply:
Why not use the [WorkflowID](https://github.com/artefactual-sdps/enduro/blob/main/internal/package_/workflow.go#L30) from the processing Workflow?  You'd have to pass it to `preprocessing` and `postprocessing` too, but it's a unique identifier that is tied to the Workflow execution, and you can set it in test as well.